Improve ansible-galaxy error for InconsistentCandidate exception

s-hertel opened the pull request:
##### SUMMARY
Fixes #75139

##### ISSUE TYPE
- Bugfix Pull Request

s-hertel committed:
Improve error for InconsistentCandidate exceptions

s-hertel committed:
changelog

s-hertel committed:
pep8

Akasurde commented on changelogs/fragments/75235-ansible-galaxy-inconsistent-candidate-error.yml:
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - ansible-galaxy - Improve error message from dependency resolution when a candidate has inconsistent requirements.
```
```suggestion
  - ansible-galaxy - Improve error message from dependency resolution when a candidate has inconsistent requirements (https://github.com/ansible/ansible/issues/75139).
```

s-hertel committed:
Update changelogs/fragments/75235-ansible-galaxy-inconsistent-candidate-error.yml

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

webknjaz commented on lib/ansible/galaxy/collection/__init__.py:
```diff
@@ -1353,3 +1354,29 @@ def _resolve_depenency_map(
+                format(
+                    req=dep_exc.candidate,
+                    dep_origin='direct request'
+                    if not parents else 'dependency of {parent!s}'.
+                    format(parent=','.join(parents))
```
How about having a whitespace after the commas?
```suggestion
                    if not parents else 'dependency of {parents!s}'.
                    format(parents=', '.join(parents))
```

webknjaz commented on test/integration/targets/ansible-galaxy-collection/tasks/install.yml:
```diff
@@ -163,6 +163,36 @@
+    error: "{{ result.stderr | regex_replace(reset_color) | regex_replace(color) | regex_replace('\\n', ' ') }}"
+    expected_error: >-
+            ERROR! Failed to resolve the requested dependencies map.
+            Got the candidate namespace3.name:1.0.0 (direct request)
```
It'd be useful to have a "dependency of" test case too.

s-hertel replied:
This caught a bug, thanks!

s-hertel committed:
Fix bug displaying parents of inconsistent candidates

Add test case for installing a collection with an inconsistent dependency version

s-hertel committed:
Join parents with ', ' instead of ','

s-hertel commented:
Thanks for the reviews!
